tests/src/test/330/scala/com/nvidia/spark/rapids/IntervalDivisionSuite.scala

```diff
@@ -0,0 +1,287 @@
+          }
+        } else if (num.isInstanceOf[Double]) {
+          if (num.equals(Double.NaN)) {
+            df.selectExpr("c1 / cast('NaN' as double)")
+          } else if (num.equals(Double.PositiveInfinity)) {
+            df.selectExpr("c1 / cast('Infinity' as double)")
+          }
+          else if (num.equals(Double.NegativeInfinity)) {
+            df.selectExpr("c1 / cast('-Infinity' as double)")
+          } else {
+            df.selectExpr(s"c1 / ${num}d")
+          }
+        } else {
+          throw new IllegalStateException("")
+        }
+    }
+
+    testBothCpuGpuExpectedException[SparkException](testCaseName + ", cv / cv",
+      e => e.getMessage.contains("ArithmeticException"),
+      spark => {
+        // Period is the external type of year-month type
+        val (data, schema) =
+          if (num.isInstanceOf[Byte]) {
+            (Seq(Row(Period.ofMonths(months), num.asInstanceOf[Byte])),
+                StructType(Seq(
+                  StructField("c1", YearMonthIntervalType()),
+                  StructField("c2", ByteType)
+                )))
+          } else if (num.isInstanceOf[Short]) {
+            (Seq(Row(Period.ofMonths(months), num.asInstanceOf[Short])),
+                StructType(Seq(
+                  StructField("c1", YearMonthIntervalType()),
+                  StructField("c2", ShortType)
+                )))
+          } else if (num.isInstanceOf[Int]) {
+            (Seq(Row(Period.ofMonths(months), num.asInstanceOf[Int])),
+                StructType(Seq(
+                  StructField("c1", YearMonthIntervalType()),
+                  StructField("c2", IntegerType)
+                )))
+          } else if (num.isInstanceOf[Long]) {
+            (Seq(Row(Period.ofMonths(months), num.asInstanceOf[Long])),
+                StructType(Seq(
+                  StructField("c1", YearMonthIntervalType()),
+                  StructField("c2", LongType)
+                )))
+          } else if (num.isInstanceOf[Float]) {
+            (Seq(Row(Period.ofMonths(months), num.asInstanceOf[Float])),
+                StructType(Seq(
+                  StructField("c1", YearMonthIntervalType()),
+                  StructField("c2", FloatType)
+                )))
+          } else if (num.isInstanceOf[Double]) {
+            (Seq(Row(Period.ofMonths(months), num.asInstanceOf[Double])),
+                StructType(Seq(
+                  StructField("c1", YearMonthIntervalType()),
+                  StructField("c2", DoubleType)
+                )))
+          } else {
+            throw new IllegalStateException("")
+          }
+
+        spark.createDataFrame(spark.sparkContext.parallelize(data), schema)
+      }
+    ) {
+      df => df.selectExpr(s"c1 / c2")
+    }
+
+    testBothCpuGpuExpectedException[SparkException](testCaseName + ", scalar / cv",
+      e => e.getMessage.contains("ArithmeticException"),
+      spark => {
+        val (data, schema) =
+          if (num.isInstanceOf[Byte]) {
```

Comment:
Nit: pattern matching would be more natural

Reply:
done